### PR TITLE
Use 0.7.0 version for golang.org/x/net to fix the vulnerability CVE-2022-41723

### PR DESCRIPTION
**Pull request by vikaspalaskar**
This temporary override is needed to address the vulnerability in golang.org/x/net which is pulled indirectly but with a version that has the vulnerability.

See [Uncontrolled Resource Consumption](https://github.com/advisories/GHSA-vvpx-j8f3-3w6h)

The only change was to run:

```
go get golang.org/x/net@v0.7.0
cd examples\xds-sotw-config-server
go get golang.org/x/net@v0.7.0
```

Signed-off-by: Vikas Palaskar <vikas.Palaskar@gmail.com>

### CODE REVIEW

**Comment by vikaspalaskar** [comment]
@envoyproxy/ratelimit-maintainers can you please review?

**Review by mattklein123 — APPROVED** [review]

File: examples/xds-sotw-config-server/go.mod [hunk 1]
```diff
@@ -13,8 +13,10 @@ require (
...
 	golang.org/x/net v0.4.0 // indirect
-	golang.org/x/sys v0.3.0 // indirect
-	golang.org/x/text v0.5.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	golang.org/x/text v0.7.0 // indirect
 	google.golang.org/genproto v0.0.0-20221118155620-16455021b5e6 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
 )
+
+replace golang.org/x/net => golang.org/x/net v0.7.0
```
Comment (arkodg) [inline examples/xds-sotw-config-server/go.mod:22]: why is this a replace

**Comment by mattklein123** [comment]
Will wait for answer to @arkodg question before merging.

File: examples/xds-sotw-config-server/go.mod (hunk 1 above)
Comment (Michael-Sinz) [reply examples/xds-sotw-config-server/go.mod:22]: The replace is there such that it replaces an indirect reference.
Something that is directly referenced has a reference to golang.org/x/net - so you can either replace all of those items with newer items that then transitively fix the reference to golang.org/x/net or you can just do the replace operation to replace that reference.

It is a powerful feature of the go build environment that can, when used carefully be very useful in multiple ways (and, as with all powerful features, can be abused too)

**Comment by vikaspalaskar** [comment]
> Will wait for answer to @arkodg question before merging.

Michael Sinz replied to the Qs. Please take a look again.

**Comment by arkodg** [comment]
@vikaspalaskar you probably just want to fetch the newer version using `go get`
```
go get golang.org/x/net@v0.7.0
go: upgraded golang.org/x/net v0.4.0 => v0.7.0
go: upgraded golang.org/x/sys v0.3.0 => v0.5.0
go: upgraded golang.org/x/text v0.5.0 => v0.7.0
🐳 ~/go-workspace/src/github.com/envoyproxy/ratelimit$ git diff
diff --git a/go.mod b/go.mod
index 3ed6672..aa17886 100644
--- a/go.mod
+++ b/go.mod
@@ -18,7 +18,7 @@ require (
        github.com/mediocregopher/radix/v3 v3.8.1
        github.com/sirupsen/logrus v1.6.0
        github.com/stretchr/testify v1.8.1
-       golang.org/x/net v0.4.0
+       golang.org/x/net v0.7.0
        google.golang.org/grpc v1.52.0
        google.golang.org/protobuf v1.28.1
        gopkg.in/yaml.v2 v2.3.0
@@ -56,6 +56,6 @@ require (
        go.opentelemetry.io/otel/sdk v1.7.0
        go.opentelemetry.io/otel/trace v1.7.0
        go.opentelemetry.io/proto/otlp v0.19.0 // indirect
-       golang.org/x/sys v0.3.0 // indirect
-       golang.org/x/text v0.5.0 // indirect
+       golang.org/x/sys v0.5.0 // indirect
+       golang.org/x/text v0.7.0 // indirect
 )

```

**Comment by vikaspalaskar** [comment]
> @vikaspalaskar you probably just want to fetch the newer version using `go get`
> 
> ```
> go get golang.org/x/net@v0.7.0
> go: upgraded golang.org/x/net v0.4.0 => v0.7.0
> go: upgraded golang.org/x/sys v0.3.0 => v0.5.0
> go: upgraded golang.org/x/text v0.5.0 => v0.7.0
> 🐳 ~/go-workspace/src/github.com/envoyproxy/ratelimit$ git diff
> diff --git a/go.mod b/go.mod
> index 3ed6672..aa17886 100644
> --- a/go.mod
> +++ b/go.mod
> @@ -18,7 +18,7 @@ require (
>         github.com/mediocregopher/radix/v3 v3.8.1
>         github.com/sirupsen/logrus v1.6.0
>         github.com/stretchr/testify v1.8.1
> -       golang.org/x/net v0.4.0
> +       golang.org/x/net v0.7.0
>         google.golang.org/grpc v1.52.0
>         google.golang.org/protobuf v1.28.1
>         gopkg.in/yaml.v2 v2.3.0
> @@ -56,6 +56,6 @@ require (
>         go.opentelemetry.io/otel/sdk v1.7.0
>         go.opentelemetry.io/otel/trace v1.7.0
>         go.opentelemetry.io/proto/otlp v0.19.0 // indirect
> -       golang.org/x/sys v0.3.0 // indirect
> -       golang.org/x/text v0.5.0 // indirect
> +       golang.org/x/sys v0.5.0 // indirect
> +       golang.org/x/text v0.7.0 // indirect
>  )
> ```

Ran the command from root and example folders. Please take a look.

**Review by arkodg — APPROVED** [review]
LGTM, thanks

**Comment by vikaspalaskar** [comment]
> mattklein123

@mattklein123 , can you please review and merge the PR ?

**Comment by mattklein123** [comment]
CI failure looks legit

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**